resources/js/app/helpers/view.js

```diff
@@ -125,18 +125,39 @@ export default {
             },
 
             acceptMimeTypes(type) {
-                if (!BEDITA.uploadMimeTypes?.[type]) {
+                if (!BEDITA.uploadConfig?.accepted?.[type]) {
                     return '';
                 }
 
-                return BEDITA.uploadMimeTypes?.[type].join(',');
+                return BEDITA.uploadConfig?.accepted?.[type].join(',');
             },
 
             checkMimeForUpload(file, objectType) {
-                /** accepted mime types by object type for file upload */
-                const mimes = BEDITA.uploadMimeTypes;
-                if (mimes?.[objectType] && !mimes[objectType].includes(file.type)) {
-                    const msg = t`File type not accepted` + `: "${file.type}". ` + t`Accepted types` + `: "${mimes[objectType].join('", "')}".`;
+                const fileType = file?.type || '';
+
+                /** forbidden mime types check */
+                const forbidden = BEDITA.uploadConfig?.forbidden?.mimetypes || [];
+                if (forbidden.includes(fileType)) {
+                    const msg = t`File type forbidden` + `: "${fileType}". ` + t`Forbidden types` + `: "${forbidden.join('", "')}".`;
+                    BEDITA.warning(msg);
+
+                    return false;
+                }
+
+                /** forbidden extensions check */
+                const extensions = BEDITA.uploadConfig?.forbidden?.extensions || [];
+                const fileExtension = fileType.replace(/(.*)\//g, '');
```

Comment:
In that way you get the second part of the mime type. 
For getting file extension I would suggest something like
```suggestion
                const fileExtension = file.name.split('.').pop();
```